src/transformState.js

```diff
@@ -5,7 +5,40 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const actionsType = {
+    add: 'addProperties',
+    remove: 'removeProperties',
+    clear: 'clear',
+  };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case actionsType.add :
+        Object.assign(state, action.extraData);
+        break;
+
+      case actionsType.remove:
+        for (const key of action.keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+        break;
+
+      case actionsType.clear:
+        for (const key in state) {
+          if (state.hasOwnProperty(key)) {
```

Comment:
Of course, this key is in the object, but you get it from the object through a loop. Remove redundant checker